include/fty_common_rest_utils_web.h

```diff
@@ -111,20 +111,25 @@ static constexpr const _WSErrors _errors = { {
     } };
 #undef HTTP_TEAPOT
 
+constexpr bool _strcmp(char const * a, char const * b)
+{
+    return *a == *b && (*a == '\0' || _strcmp(a + 1, b + 1));
```

Comment:
Can a `char const *` be NULL? If yes, here you get a deref error.
Also, for sake of comparison, are two NULL arguments considered equal? 
What about a NULL and an empty string (where `arg[0]=='\0'`)?

Reply:
Also as Clement mentioned, here you recurse for every character, so for strings of any considerable length you build a huge stack to compare them.

Notably, what you made here should be named `streq()` since it says whether the two strings are identical, not how they collate alphabetically (which in turn should be locale-aware, probably UTF-8 aware, etc. at least in general case if we were to use this routine elsewhere - and in some component we did already have several routines dealing with UTF-8 specifics so this probably would then belong there as well).

Reply:
from doc of the strcmp... 
```The behavior is undefined if lhs or rhs are not pointers to null-terminated byte strings.```
So, the function should fail in case of nullptr.

Reply:
In case of utf-8 we shouldn't use strcmp at all, because this should be byte string.

Reply:
Any UTF-8 string can be compared with `strcmp` for strict equality since they do not embed NULL characters and has been designed to safely interoperate with ASCII-only software, unlike UTF-16. However, this will not cover _semantically_ equal strings that are unnormalized or normalized with different algorithms.

Reply:
About the code itself: this will invoke undefined behavior if the `a` string is longer than the `b` string, since there is no NULL check on the `b` string to prevent a buffer overrun. Also, `strcmp` normally returns an `int` enables three-way comparisons, but I suppose this is acceptable for our purposes.

Reply:
Wait, scratch that, the NULL character in `b` will not compare equal and will stop the recursion. S'all good (need more sleep).

Reply:
> About the code itself: this will invoke undefined behavior if the `a` string is longer than the `b` string, since there is no NULL check on the `b` string to prevent a buffer overrun. Also, `strcmp` normally returns an `int` enables three-way comparisons, but I suppose this is acceptable for our purposes.

I'm sorry, but one thing to my pardon is that this is as null terminating string, than in case if b is shorter than then a == 0 will finish  :)

Reply:
I slightly changed the function. :)